concept-pwa/src/main/java/mozila/components/concept/pwa/PwaCore.kt

```diff
@@ -0,0 +1,114 @@
+package mozila.components.concept.pwa
+
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.OnLifecycleEvent
+import android.graphics.BitmapFactory
+import android.util.Log
+import kotlinx.coroutines.experimental.CommonPool
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.launch
+import kotlinx.coroutines.experimental.withContext
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+import java.lang.ref.WeakReference
+import java.net.MalformedURLException
+import java.net.URL
+
+interface PwaViewContract {
+    fun notifyPwaReady(jsonObject: PwaModel)
+}
+
+private const val TAG: String = "PwaCore"
+
+
```

Comment:
\[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1